modules/standard/Types.chpl

```diff
@@ -201,68 +312,105 @@ proc chpl__maxIntTypeSameSign(type t) type {
-/* Returns `true` if the argument is a value of one the following types:
-`int`, `uint`, `real`, `imag`, `complex`. */
-proc isNumericValue(e)   param  return isNumericType(e.type);
-/* Returns `true` if the argument is a value of primitive type. */
-proc isPrimitiveValue(e) param  return isPrimitiveType(e.type);
+
 /* Returns `true` if the argument is a `enum` value. */
 proc isEnumValue(e)      param  return isEnumType(e.type);
-/* Returns `true` if the argument is a `nothing` value (i.e., `none`) */
-proc isNothingValue(e)   param return isNothingType(e.type);
-//Defined elsewhere:
-// isTupleValue
-// isHomogeneousTupleValue
-pragma "no doc"
+
+/* Returns `true` if the argument is a class value */
 proc isClassValue(e)              param return isClassType(e.type);
 
+/* Returns `true` if the argument is a record value */
+proc isRecordValue(e)    param  return isRecordType(e.type);
+
+/* Returns `true` if the argument is a union value */
+proc isUnionValue(e)     param  return isUnionType(e.type);
+
 pragma "no doc"
-proc isOwnedClassType(type t:_owned)         param return true;
-pragma "no doc"
-proc isOwnedClassType(type t)                param return false;
+proc isTupleValue(x: _tuple) param return true;
+/* Returns `true` if the argument is a tuple value */
+proc isTupleValue(x) param return false;
+
 pragma "no doc"
-proc isSharedClassType(type t:_shared)       param return true;
+proc isRangeValue(r: range(?)) param  return true;
+/* Return true if argument ``r`` is a range, false otherwise */
+proc isRangeValue(r)           param  return false;
+
 pragma "no doc"
-proc isSharedClassType(type t)               param return false;
+proc isDomainValue(e: domain) param  return true;
+/* Return true if ``e`` is a domain. Otherwise return false. */
+proc isDomainValue(e)         param  return false;
+
 pragma "no doc"
-proc isUnmanagedClassType(type t:unmanaged)  param return true;
+proc isArrayValue(e: []) param  return true;
+/* Return true if ``e`` is an array. Otherwise return false. */
+proc isArrayValue(e)     param  return false;
+
 pragma "no doc"
-proc isUnmanagedClassType(type t)            param return false;
+proc isDmapValue(e: _distribution) param  return true;
+/* Return true if ``e`` is a domain map. Otherwise return false. */
+proc isDmapValue(e)                param  return false;
+
 pragma "no doc"
-proc isBorrowedClassType(type t:borrowed)    param return true;
+proc isSyncValue(x : sync) param  return true;
+/* Returns `true` if the argument is a sync value */
+proc isSyncValue(x)       param  return false;
+
 pragma "no doc"
-proc isBorrowedClassType(type t)             param return false;
+proc isSingleValue(x : single) param  return true;
+/* Returns `true` if the argument is a single value */
+proc isSingleValue(x)         param  return false;
 
+/* Returns `true` if the argument is an atomic value */
+proc isAtomicValue(e)    param  return isAtomicType(e.type);
+
+pragma "no doc"
```

Comment:
Did you mean to leave this undocumented?  It looks like there's an `isHomogeneousTupleType` that is documented